cmd/setup-etcd-environment/run.go

```diff
@@ -50,24 +49,28 @@ func runRunCmd(cmd *cobra.Command, args []string) error {
 		return errors.New("--discovery-srv cannot be empty")
 	}
 
-	ip, err := ipAddrForIf(runOpts.ifName)
+	ips, err := ipAddrs()
 	if err != nil {
 		return err
 	}
-	glog.Infof("ip addr is %s", ip)
 
 	var dns string
-	if err := wait.PollImmediate(1*time.Minute, 5*time.Minute, func() (bool, error) {
-		found, err := reverseLookupSelf("etcd-server-ssl", "tcp", runOpts.discoverySRV, ip)
-		if err != nil {
-			glog.Errorf("error looking up self: %v", err)
-			return false, nil
-		}
-		if found != "" {
-			dns = found
-			return true, nil
+	var ip string
+	if err := wait.PollImmediate(30*time.Second, 5*time.Minute, func() (bool, error) {
+		for _, cand := range ips {
```

Comment:
This looks good just so I understand why did we reduce the interval here to 30s?

Reply:
the discovery is in the hot path of the bootstrap, I want us to be conservative as dns propagation is slow... and 1 minute is a little long for a poll.